source/Calamari.Common/Features/Packages/TarPackageExtractor.cs

```diff
@@ -25,58 +23,61 @@ public TarPackageExtractor(ILog log)
 
         public int Extract(string packageFile, string directory)
         {
+            EnsureTargetDirectoryExists(directory);
+
             var files = 0;
-            using (var inStream = new FileStream(packageFile, FileMode.Open, FileAccess.Read))
+            using var inStream = new FileStream(packageFile, FileMode.Open, FileAccess.Read);
+            var compressionStream = GetCompressionStream(inStream);
+            try
             {
-                var compressionStream = GetCompressionStream(inStream);
-                try
-                {
-                    using (var reader = TarReader.Open(compressionStream, new ReaderOptions { ArchiveEncoding = new ArchiveEncoding { Default = Encoding.UTF8 } }))
-                    {
-                        while (reader.MoveToNextEntry())
-                        {
-                            ProcessEvent(ref files, reader.Entry);
-                            ExtractEntry(directory, reader);
-                        }
-                    }
-                }
-                finally
+                using var reader = TarReader.Open(compressionStream, new ReaderOptions { ArchiveEncoding = new ArchiveEncoding { Default = Encoding.UTF8 } });
+                while (reader.MoveToNextEntry())
                 {
-                    if (compressionStream != inStream)
-                        compressionStream.Dispose();
+                    ProcessEvent(ref files, reader.Entry);
+                    ExtractEntry(directory, reader);
                 }
             }
+            finally
+            {
+                if (compressionStream != inStream)
+                    compressionStream.Dispose();
+            }
 
             return files;
         }
 
+        static void EnsureTargetDirectoryExists(string directory) => Directory.CreateDirectory(directory);
```

Comment:
There was a change in SharpCompress `0.29` that it now verifies that the target directory exists before extraction, so we now verify the target directory exists :)